elastic/kibana
# [Fleet] refactor auto upgrade package policies logic

## Summary

Fixes https://github.com/elastic/kibana/issues/121639

**Refactored logic of auto upgrade package policies:**
- querying installed packages with `keep_policies_up_to_date` flag, then
- query package policies where `version != package version`
- filter out package policies where `version >= package version`
- run dry run and then run upgrade if no errors

**Separated auto upgrade package policy logic from ensure preconfigured policies logic, so that it is no longer called when invoking API `POST /api/fleet/setup/preconfiguration`**

**To verify (existing functionality):**
- install old version of system package
```
POST http://elastic:changeme@localhost:5620/api/fleet/epm/packages/system-1.9.0
kbn-xsrf: kibana

{ "force": true }
```
- add system integration
- switch keep policies up to date on (integration settings page)
- update package to latest without policies
- navigate to Fleet app (or POST /api/fleet/setup)
- verify that system package policy is upgraded


### Checklist
